### PR TITLE
Add SNAP (CA, 22) parameters

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
This should get us most of the way to coverage of SNAP for California in 2022. I've added a few placeholders for remaining SNAP logic (min benefit), though @nmrodelo these are just placeholders and your PR is a lot closer! I've added the utility allowance logic (standard/limited/telephone) and the rates for 2022 as per #388. @MaxGhenis how's this look?

### CODE REVIEW

File: openfisca_us/variables/usda/snap/snap.py [hunk 1]
```diff
@@ -11,8 +11,12 @@ class snap(Variable):
...
     def formula(spm_unit, period):
-        max_benefit = spm_unit("snap_max_benefit", period)
-        expected_contribution = spm_unit(
-            "snap_expected_contribution_towards_food", period
+        # Federal SNAP rules are defined in U.S.C Title 7, Chapter 51, which also
+        # defines state powers to modify the rules.
+        eligible = spm_unit("is_snap_eligible", period)
+        max_benefit = max_(
+            spm_unit("snap_max_benefit", period),
+            spm_unit("snap_min_benefit", period),
```
Comment (MaxGhenis) [inline openfisca_us/variables/usda/snap/snap.py:19]: Doesn't the minimum benefit belong in the return statement's `max_` call?

**Review by MaxGhenis — COMMENTED** [review]
Could you add the California document as the source for updated 2022 federal parameters? For now at least, until we get a federal document.

https://www.cdss.ca.gov/Portals/9/Additional-Resources/Letters-and-Notices/ACINs/2021/I-78_21.pdf

File: openfisca_us/parameters/usda/snap/homeless_shelter_deduction.yaml [hunk 2]
```diff
@@ -2,10 +2,11 @@ description: SNAP Homeless Shelter Deduction Parameters
 
 values:
   2021-01-01: 143
+  2022-10-01: 159.73
```
[the remark this answers is not in the document]
Comment (nikhilwoodruff) [reply openfisca_us/parameters/usda/snap/homeless_shelter_deduction.yaml:5]: Yeah- added some references and corrected

File: openfisca_us/variables/usda/snap/income/deductions/shelter_expense.py [hunk 3]
```diff
@@ -83,3 +83,90 @@ def formula(spm_unit, period, parameters):
...
+
+
+class SNAPUTtilityAllowanceType(Enum):
+    SUA = "Standard Utility Allowance"
+    LUA = "Limited Utility Allowance"
+    TUA = "Telephone Utility Allowance"
+    NONE = "None"
+
+
+class has_heating_cooling_expense(Variable):
```
[the remark this answers is not in the document]
Comment (nikhilwoodruff) [reply openfisca_us/variables/usda/snap/income/deductions/shelter_expense.py:95]: I think it's cleaner to collect this for mandatory states, and override with the actual amount if the parameter "mandatory SUA" for the state is false.

File: openfisca_us/variables/usda/snap/income/income.py [hunk 4]
```diff
@@ -21,8 +21,9 @@ class snap_net_income(Variable):
     documentation = "Final net income, after all deductions"
     label = "SNAP net income"
     unit = "currency-USD"
+    reference = "https://www.law.cornell.edu/uscode/text/7/2014"
```
[the remark this answers is not in the document]
Comment (nikhilwoodruff) [reply openfisca_us/variables/usda/snap/income/income.py:24]: Moved to `snap_expected_contribution`, which references 2017(a) (the section actually specifying rounding down)

File: openfisca_us/variables/usda/snap/snap.py (hunk 1 above)
Comment (nikhilwoodruff) [reply openfisca_us/variables/usda/snap/snap.py:19]: Ah yep

**Pull request merged**